src/components/contextual/pages/vebal/voting-utils.ts

```diff
@@ -0,0 +1,67 @@
+import { VotingPool } from '@/composables/queries/useVotingPoolsQuery';
+import { FNumFormats, numF } from '@/composables/useNumbers';
+import { oneSecondInMs, toUtcTime } from '@/composables/useTime';
+import { WEIGHT_VOTE_DELAY } from '@/constants/gauge-controller';
+import { bnum, isSameAddress, scale } from '@/lib/utils';
+import { BigNumber } from '@ethersproject/bignumber';
+import { differenceInWeeks, format } from 'date-fns';
+
+/*
+ * Common pure functions used by different composables in the veBAL voting feature
+ */
+export function isGaugeNew(pool: VotingPool): boolean {
+  const addedTimestamp = pool.gauge.addedTimestamp;
+  if (!addedTimestamp) return false;
+  return differenceInWeeks(Date.now(), addedTimestamp * oneSecondInMs) < 2;
+}
+
+export function isGaugeExpired(
+  expiredGauges: readonly string[] | undefined,
+  gaugeAddress: string
+): boolean {
+  if (!expiredGauges) return false;
+  return !!expiredGauges.some(item => isSameAddress(gaugeAddress, item));
+}
+
+export function isPoolExpired(pool: VotingPool) {
+  return pool.gauge.isKilled;
+}
+
+export function hasOnlyExpiredPools(
+  votingGauges: string[],
+  expiredGauges?: readonly string[]
+) {
+  if (!expiredGauges) return false;
+  return votingGauges.every(gaugeAddress =>
+    isGaugeExpired(expiredGauges, gaugeAddress)
+  );
+}
+
+// Vote weight is saved as basis points (bps) onchain (i.e. 20% is stored as 2000)
+// so we use this method to work with shares in the UI
+export function bpsToShares(weight: string): string {
```

Comment:
We could move this helpers to `useNumbers` file but I preferred to keep them closer to the voting domain for higher cohesion.